### ISSUE TITLE
Flaky test: TestParseWords

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#23517]
Looks like it's mainly failing on ZFS, but not sure if that's related;

https://jenkins.dockerproject.org/job/Docker%20Master/label=ubuntu%20&&%20zfs/5741/console

```
08:39:17 --- FAIL: TestParseWords (0.00s)
08:39:17 	parser_test.go:124: length check failed. input: foo\ bar, expect: [foo\ bar], output: [foo\ bar]
08:39:17 FAIL
```

**Comment by tonistiigi** [comment]
Possible cause: #23999

**Comment by dnephin** [comment]
Fixed by #25068
